pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -546,8 +546,12 @@ func (s *discovery) updateServicesCPUStats() error {
 		return fmt.Errorf("could not get global CPU time: %w", err)
 	}
 
-	for pid, serviceInfo := range s.cache {
-		_ = updateCPUCoresStats(pid, serviceInfo, s.lastGlobalCPUTime, globalCPUTime)
+	for i := range services {
+		service := &services[i]
+		serviceInfo := s.cache[int32(service.PID)]
```

Comment:
Don't we need a .., ok := ... check for safety?  Isn't there a (theoretical) risk of the service disappearing from the cache if two instances of the handler run and one gets delayed?